components/governance/org.wso2.carbon.governance.taxonomy.ui/src/main/resources/web/taxonomy/js/form.js

```diff
@@ -0,0 +1,623 @@
+                    a.push({name: n, value: options.data[n]});
+            }
+        }
+
+        // give pre-submit callback an opportunity to abort the submit
+        if (options.beforeSubmit && options.beforeSubmit(a, this, options) === false) {
+            log('ajaxSubmit: submit aborted via beforeSubmit callback');
+            return this;
+        }
+
+        // fire vetoable 'validate' event
+        this.trigger('form-submit-validate', [a, this, options, veto]);
+        if (veto.veto) {
+            log('ajaxSubmit: submit vetoed via form-submit-validate trigger');
+            return this;
+        }
+
+        var q = $.param(a);
+
+        if (options.type.toUpperCase() == 'GET') {
+            options.url += (options.url.indexOf('?') >= 0 ? '&' : '?') + q;
+            options.data = null;  // data is null for 'get'
+        }
+        else
+            options.data = q; // data is the query string for 'post'
+
+        var $form = this, callbacks = [];
+        if (options.resetForm) callbacks.push(function () {
+            $form.resetForm();
+        });
+        if (options.clearForm) callbacks.push(function () {
+            $form.clearForm();
+        });
+
+        // perform a load on the target only if dataType is not provided
+        if (!options.dataType && options.target) {
+            var oldSuccess = options.success || function () {
+                };
+            callbacks.push(function (data) {
+                $(options.target).html(data).each(oldSuccess, arguments);
+            });
+        }
+        else if (options.success)
+            callbacks.push(options.success);
+
+        options.success = function (data, status) {
+            for (var i = 0, max = callbacks.length; i < max; i++)
+                callbacks[i].apply(options, [data, status, $form]);
+        };
+
+        // are there files to upload?
+        var files = $('input:file', this).fieldValue();
+        var found = false;
+        for (var j = 0; j < files.length; j++)
+            if (files[j])
+                found = true;
+
+        // options.iframe allows user to force iframe mode
+        if (options.iframe || found) {
+            // hack to fix Safari hang (thanks to Tim Molendijk for this)
+            // see:  http://groups.google.com/group/jquery-dev/browse_thread/thread/36395b7ab510dd5d
+            if ($.browser.safari && options.closeKeepAlive)
+                $.get(options.closeKeepAlive, fileUpload);
+            else
+                fileUpload();
+        }
+        else
+            $.ajax(options);
+
+        // fire 'notify' event
+        this.trigger('form-submit-notify', [this, options]);
+        return this;
+
+
+        // private function for handling file uploads (hat tip to YAHOO!)
+        function fileUpload() {
+            var form = $form[0];
+
+            if ($(':input[@name=submit]', form).length) {
+                alert('Error: Form elements must not be named "submit".');
```

Comment:
Thank you Sameera for mentioning it, 
Yes, Its a third party library. I have added license code because , we have added a plugin to check all our license hearers. So plugin will not fail. (https://wso2.org/jira/browse/REGISTRY-2963)